pymeasure/instruments/ametek/ametek7270.py

```diff
@@ -137,94 +149,127 @@ class Ametek7270(Instrument):
-                                  """ Reads the input value of ADC1 in Volts """
+                                  """ Reads the input value of ADC1 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     adc2 = Instrument.measurement("ADC. 2",
-                                  """ Reads the input value of ADC2 in Volts """
+                                  """ Reads the input value of ADC2 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     adc3 = Instrument.measurement("ADC. 3",
-                                  """ Reads the input value of ADC3 in Volts """
+                                  """ Reads the input value of ADC3 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     adc4 = Instrument.measurement("ADC. 4",
-                                  """ Reads the input value of ADC4 in Volts """
+                                  """ Reads the input value of ADC4 in Volts """,
+                                  get_process=check_read_not_empty,
                                   )
     id = Instrument.measurement("ID",
                                 """ Reads the instrument identification """,
-                                cast=str
+                                cast=str,
                                 )
 
     def __init__(self, adapter, name="Ametek DSP 7270",
                  read_termination='\x00',
                  write_termination='\x00',
-                 **kwargs):
+                 **kwargs,
+                 ):
 
         super().__init__(
             adapter,
             name,
             read_termination=read_termination,
             write_termination=write_termination,
-            **kwargs
+            **kwargs,
         )
 
+    def check_errors(self):
+        """mandatory to be used for property setter
+
+        The Ametek protocol expect the default null character to be read to check the property
+        has been correctly set. With default termination character set as Null character,
+        this turns out as an empty string to be read.
+        """
+        if self.read() == '':
+            return []
+        else:
+            return ['Incorrect return from previously set property']
+
+    def ask(self, command, query_delay=0):
+        """Usually the properties use the `values` method that adds a strip call,
+        however several methods use directly the result from ask to be casted into some other types.
+        It should therefore also add the strip here
+        """
+        return super().ask(command, query_delay).strip()
+
+    def set_reference_mode(self, mode: int = 0):
+        """Set the instrument in Single, Dual or harmonic modes
+        :param mode: the integer specifying the mode: 0 for Single, 1 for Dual harmonic and 2 for Dual
+        reference

```

Comment:
Small improvements:
```suggestion
        """Set the instrument in Single, Dual or harmonic mode.

        :param mode: the integer specifying the mode: 0 for Single, 1 for Dual harmonic, and 2 for
            Dual reference.

```
Reasons:
- The first line is a sentence in imperative mode, which should end with a period
- Between the summary line (first line) and additional text, should be an empty line.